src/VideoStreamer/VideoStreamer.cpp

```diff
@@ -0,0 +1,288 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+#include "VideoStreamerInternal.h"
+#include <iostream>
+#pragma comment(lib, "avformat.lib")
+using namespace winrt;
+#ifdef TIGHT_LATENCY_CONTROL
+    uint32_t g_dropCount;
+#endif
+struct GUIDComparer
+{
+    bool operator()(const GUID& Left, const GUID& Right) const
+    {
+        // comparison logic goes here
+        return memcmp(&Left, &Right, sizeof(Right)) < 0;
+    }
+};
+std::map <GUID, AVCodecID, GUIDComparer> g_CodecMapMFtoFF =
+{ {MFVideoFormat_H264, AV_CODEC_ID_H264},
+  {MFVideoFormat_MPEG2, AV_CODEC_ID_MPEG2VIDEO}
+};
+
+bool VideoStreamerFFmpeg::s_FFmpegInitDone = false;
+
+void VideoStreamerBase::WritePacket(IMFSample* pSample)
+{
+    winrt::check_hresult(m_spSinkWriter->WriteSample(0, pSample));
+}
+
+VideoStreamerFFmpeg::~VideoStreamerFFmpeg()
+{
+    for (auto& avc : m_aAvfctx)
+    {
+        avformat_free_context(avc.second);
+    }
+    m_aAvfctx.clear();
+}
+
+void VideoStreamerFFmpeg::InitFFmpeg()
+{
+    if (!s_FFmpegInitDone)
+    {
+        //avcodec_register_all();
+        //av_register_all();

```

Comment:
remove?

Reply:
as well as line 53 and 55

---
In reply to: [403716284](https://github.com/ms-iot/ros_win_camera/pull/4#discussion_r403716284) [](ancestors = 403716284)